### DISCUSSION TITLE
Livewire file upload file name too long

### DISCUSSION THREAD

**Discussion by oxytoxin** [livewire/livewire#2813] [1. Help] [answered]
![image](https://user-images.githubusercontent.com/61694153/116951754-49166b80-acbb-11eb-955e-a3f07665b853.png)
What should I do to shorten the filename that livewire uses to store temporary files?

**Comment by ZerglingGo** [comment, marked as answer]
Livewire currently doesn't seem to handle long filenames. (I think that was related to getting the original filename.)

https://github.com/livewire/livewire/blob/6c45c16952a23447474079613d87af38669b13f8/src/TemporaryUploadedFile.php#L126-L133

Until the fixing problem, you must shorten the original filename before uploading.
I'll create an issue on this problem.

**Reply by oxytoxin** [reply]
Yes, I saw this too. I was using filepond, so I ended up using its onerror callback to listen to errors and check for the filename length so I can notify the user. The only thing left is that livewire still throws the error so sentry.io still picks it up.

**Reply by ZerglingGo** [reply]
If you need only "Don't report" options,  How about filtering in `App\Exceptions\Handler`?

```php
public function register()
{
    $this->reportable(function (Throwable $e) {
        if (\Str::endsWith($e->getMessage(), 'File name too long')) { // Filtering that error message
            return false;
        }
    });
}
```

You can disable logging of only those errors in this way.

**Reply by oxytoxin** [reply]
Wow, thank you for this. I will try this later. Cheers!

**Reply by calebporzio** [reply]
So either of you feel up to fixing this in a PR?

**Reply by ZerglingGo** [reply]
I will do PR fixed codes.
